### PR TITLE
Add alias target "Zycore::Zycore"

### PR DESCRIPTION
**Pull request by dg0yt**
To offer the same public target regardless of zycore-c is used as a subproject or via `find_package`.
This should be complemented by Zydis linking "Zycore::Zycore".

(Noticed in reviewing a vcpkg port update.)

### CODE REVIEW

**Comment by flobernd** [comment]
Hi @dg0yt, thanks for your contribution. Happy to merge this, but could you please explain me the why this is required? I'm not a CMake expert and don't quite see where the `Zycore::Zycore` comes from. Is that the way `find_package` exposes the lib, if no submodule is used?

**Review by flobernd — APPROVED** [review]

**Comment by dg0yt** [comment]
> Is that the way `find_package` exposes the lib, if no submodule is used?

Yes. The plain `Zycore` target doesn't exist after `find_package(Zycore)`. The `Zycore::` prefix is coming from the `NAMESPACE` given in the `install(EXPORT ...)` command. 

> why this is required?

Namespaces are not required, but here it is already added. And from the perspective of the consuming projects (Zydis), namespaces are a good thing: When CMake sees `::`, it knows that the identifier shall be a target. It can create a proper error message at CMake generation time. (A::A doesn't exist. Did you forget find_package(A)?)") Plain names are considered "lib specs" if they are not targets, i.e. the linker will search for the name. If the user forgot `find_package`, she/he will see an error only at the end of the build, with no good diagnostics.

Basically, Zycore/Zydis form the text book example how it should be used.

**Comment by flobernd** [comment]
Thanks a lot for the explanation @dg0yt 🙂 Merging this right now. Feel free to update the submodule in the Zydis PR so that we can merge this as well.
